### PR TITLE
M0127CC-611 data formats tutorial

### PR DESCRIPTION
**Pull request by fpinel-microej**

### CODE REVIEW

**Comment by frivieremicroej** [comment]
Link: https://docs.microej.com/en/feature-m0127cc-611_data_formats_tutorial/

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+Intended Audience
+=================
+
+The audience for this document is Application engineers who want to implement data serialization.
+In addition, this tutorial should be of interest to software architects how are looking for a suitable data format for their use case.
+
+XML
+===
+
+`XML (EXtensible Markup Language) <https://en.wikipedia.org/wiki/XML>`_ is used to describe data and text. It allows flexible development of user-defined document types. The format is robust, non-proprietary, persistent and is verifiable for storage and transmission. To parse this data format, several XML Pull parsers are available, such as `KXmlParser <http://kxml.org/>`_ from the Java community, which has been integrated to MicroEJ as a MMM dependency.
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:15]:
```suggestion
`XML (EXtensible Markup Language) <https://en.wikipedia.org/wiki/XML>`_ is used to describe data and text. It allows flexible development of user-defined document types. The format is robust, non-proprietary, persistent and is verifiable for storage and transmission. To parse this data format, several XML Pull parsers are available, such as `KXmlParser <http://kxml.org/>`_ from the Java community, which has been integrated to MicroEJ Central Repository.
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+XML Parser Use In MicroEJ SDK
+-----------------------------
+
+The `XML API Module <https://repository.microej.com/artifacts/org/kxml2/kxml2/>`_ must be added to the :ref:`module.ivy <mmm_module_description>` of the MicroEJ 
+Application project in order to allow access to the Device library.
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:21]:
```suggestion
Application project in order to allow access to the KXML library.
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+::
+
+	<dependency org="org.kxml2" name="kxml2" rev="2.3.2"/>
+
+Example Of Use
+--------------
+
+This MicroEJ project presents how to use XML data exchange for your Java application developed with MicroEJ. It details how to use the `KXmlParser <http://kxml.org/>`_ module.
+
+The example parses a short poem written in XML and prints the result on the system out. The MicroEJ project runs on all built-in MicroEJ Java platforms (simulator and evaluation kits).
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:32]:
```suggestion
The example parses a short poem written in XML and prints the result on the system out. The MicroEJ project can run on any MicroEJ Platform (no external dependencies).
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		<l>Sugar is sweet,</l>
+		<l>And I love you.</l>
+	</poem>
+
+JSON
+====
+
+As described on the `JSON official site <http://json.org/>`_, JSON (JavaScript Object Notation) is a lightweight data-interchange format. It is widely used in many fields for example as a mean of data serialization in lightweight web services such as REST, for server interrogation in Ajax for dynamic webpage building, or even databases.
+
+Due to its lightweight nature and its simplified syntax compared to XML, it is quite natural to use it in embedded systems. JSON is easily readable by humans but what about computers? That is where the official JSON page comes in handy by providing a rather impressive list of JSON parsers. For our convenience a Java mobile edition (ME) of a JSON parser is available.
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:58]:
```suggestion
JSON is easily readable by humans compared to XML. To parse this data format, several JSON parsers are available on the `official JSON page<http://json.org/>`_, such as `JSON ME`, which has been integrated to MicroEJ Central Repository.
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+JSON Parser Use In MicroEJ SDK
+------------------------------
+
+The `JSON API Module <https://repository.microej.com/artifacts/org/json/me/json/>`_ must be added to the :ref:`module.ivy <mmm_module_description>` of the MicroEJ 
+Application project in order to allow access to the Device library.
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:64]:
```suggestion
Application project in order to allow access to the JSON library.
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+				// a getJSONObject() or a getJSONArray() failed
+				// or the parsing failed
+				e.printStackTrace();
+			}
+
+		}
+
+	}
+
+The execution of this example on the simulated Java platform should print the following trace:
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:186]: MicroEJ Simulator (use glossary terms)
```suggestion
The execution of this example on the MicroEJ Simulator should print the following trace:
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+::
+
+	=============== [ Initialization Stage ] ===============
+	=============== [ Launching SimJPF ] ===============
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:191]: ### 
```suggestion
	=============== [ Launching Simulator] ===============
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+The example below will parse the file, browse the resulting data structure (``org.json.me.JSONObject``) and print the value of the ``menuitem`` JSON array.
+
+.. code:: Java
+
+	/*
+	 * Java
+	 *
+	 * Copyright 2014-2020 MicroEJ Corp. All rights reserved.
+	 */
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:115]:
```suggestion
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+::
+
+	<?xml version="1.0" encoding="UTF-8"?>
+	<!--
+		XML
+		
+		Copyright 2014-2020 MicroEJ Corp. All rights reserved.
+		Use of this source code is governed by a BSD-style license that can be found with this software.
+	-->
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:44]:
```suggestion
```

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 10]
```diff
@@ -0,0 +1,205 @@
+Evaluate Data Serialization Formats
+'''''''''''''''''''''''''''''''''''
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:2]: use same titles seperators order than other tutorials

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+	<dependency org="org.kxml2" name="kxml2" rev="2.3.2"/>
+
+Example Of Use
+--------------
+
+This MicroEJ project presents how to use XML data exchange for your Java application developed with MicroEJ. It details how to use the `KXmlParser <http://kxml.org/>`_ module.
+
+The example parses a short poem written in XML and prints the result on the system out. The MicroEJ project runs on all built-in MicroEJ Java platforms (simulator and evaluation kits).
+
+All source code is available on `MicroEJ GitHub <https://github.com/MicroEJ/Example-XML>`_ and you can find a free trial version of MicroEJ SDK `here <https://developer.microej.com/get-started/>`_.
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:34]:
```suggestion
All source code is available at https://github.com/MicroEJ/Example-XML. 
```
show the link when this is necessary to be understood (think about PDF)

File: Tutorials/tutorialEvaluateDataSerializationFormats.rst [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+		Use of this source code is governed by a BSD-style license that can be found with this software.
+	-->
+	<poem xmlns="http://www.megginson.com/ns/exp/poetry">
+		<title>Roses are Red</title>
+		<l>Roses are red,</l>
+		<l>Violets are blue;</l>
+		<l>Sugar is sweet,</l>
+		<l>And I love you.</l>
+	</poem>
+
```
Comment (frivieremicroej) [inline Tutorials/tutorialEvaluateDataSerializationFormats.rst:52]: I suggest to add the exepected Simulator output as for JSON tuto

File: Tutorials/tutorialExploreDataSerializationFormats.rst [hunk 13]
```diff
@@ -0,0 +1,249 @@
+Explore Data Serialization Formats
+==================================
+
+This tutorial highlights some data serialization formats that are provided on MicroEJ Central Repository and their usage through basic code samples.
+
+Intended Audience
+-----------------
+
+The audience for this document is Application engineers who want to implement data serialization.
+In addition, this tutorial should be of interest to software architects how are looking for a suitable data format for their use case.
```
Comment (gjadi-mej) [inline Tutorials/tutorialExploreDataSerializationFormats.rst:10]:
```suggestion
In addition, this tutorial should be of interest to software architects whow are looking for a suitable data format for their use case.
```

File: Tutorials/tutorialExploreDataSerializationFormats.rst [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+
+XML
+---
+
+`XML (EXtensible Markup Language) <https://en.wikipedia.org/wiki/XML>`_ is used to describe data and text. It allows flexible development of user-defined document types. The format is robust, non-proprietary, persistent and is verifiable for storage and transmission. To parse this data format, several XML Pull parsers are available, such as `KXmlParser <http://kxml.org/>`_ from the Java community, which has been integrated to MicroEJ Central Repository.
```
Comment (gjadi-mej) [inline Tutorials/tutorialExploreDataSerializationFormats.rst:15]:
```suggestion
`XML (EXtensible Markup Language) <https://en.wikipedia.org/wiki/XML>`_ is used to describe data and text. It allows flexible development of user-defined document types. The format is robust, non-proprietary, persistent and is verifiable for storage and transmission. To parse this data format, the XML Pull parser `KXmlParser <http://kxml.org/>`_ from the Java community has been integrated to MicroEJ Central Repository.
```

AFAIK only one parser is available on MEJ Central Repo

File: Tutorials/tutorialExploreDataSerializationFormats.rst [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+	Characters:    "\n"
+	End element:   {http://www.megginson.com/ns/exp/poetry}poem
+	=============== [ Completed Successfully ] ===============
+
+	SUCCESS
+
+JSON
+----
+
+As described on the `JSON official site <http://json.org/>`_, JSON (JavaScript Object Notation) is a lightweight data-interchange format. It is widely used in many fields for example as a mean of data serialization in lightweight web services such as REST, for server interrogation in Ajax for dynamic webpage building, or even databases.
```
Comment (gjadi-mej) [inline Tutorials/tutorialExploreDataSerializationFormats.rst:104]:
```suggestion
As described on the `JSON official site <http://json.org/>`_, JSON (JavaScript Object Notation) is a lightweight data-interchange format. It is widely used in many applications such as:

- as a mean of data serialization for lightweight web services such as REST
- for server interrogation in Ajax to build dynamic webpages
- or even databases.
```

**Pull request merged**